tests/test_types.py

```diff
@@ -955,6 +957,32 @@ class Model(BaseModel):
         Model(v=b'1')
 
 
+def test_strict_int():
+    class Model(BaseModel):
+        v: StrictInt
+
+    assert Model(v=123456).v == 123456
+
+    with pytest.raises(ValidationError):
+        Model(v='123456')
+
+    with pytest.raises(ValidationError):
+        Model(v=3.14159)
+
+
+def test_strict_float():
+    class Model(BaseModel):
+        v: StrictFloat
+
+    assert Model(v=3.14159).v == 3.14159
+
+    with pytest.raises(ValidationError):
+        Model(v='3.14159')
+
+    with pytest.raises(ValidationError):
+        Model(v=123456)
```

Comment:
surely integers should be valid to a `float` field? The zen of python 3 (as far as I know) is "anywhere that a float works, an int should work too".

It feels very "python 2" to have to do `Model(v=123456.0)`.

What do you think?

Reply:
As far as only Python is concerned I would agree but as a parsing library pydantic lives at the boundaries between python code and other systems that need to communicate via serialized data.

Most often it should be the case that all integers can be treated as floats especially with JSON but I don't want to make that decision for every use case out there based on just mine.

Reply:
okay, I guess this is clearest. 

Can you add some notes/gotchas to docs, noting things like:
* although `bool` is a subcalss of `int` in python, they're not allowed
* `int` is not allowed to a `float` field

any others?

Reply:
None I can think of.

Reply:
> As far as only Python is concerned I would agree but as a parsing library pydantic lives at the boundaries between python code and other systems that need to communicate via serialized data.
> 
> Most often it should be the case that all integers can be treated as floats especially with JSON but I don't want to make that decision for every use case out there based on just mine.

I agree that the current approach is the clearest. In any case, it will be easy for someone to re-implement it the other way around if he/she wishes so in her own code.